src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/Process_StreamTests.cs

```diff
@@ -63,25 +64,31 @@ public void Process_SyncOutputStream()
         public void Process_AsyncOutputStream()
         {
             {
+                AutoResetEvent are = new AutoResetEvent(false);
+                int expectedLines = 2;
                 StringBuilder sb = new StringBuilder();
                 Process p = CreateProcessStream();
                 p.StartInfo.RedirectStandardOutput = true;
-                p.OutputDataReceived += (s, e) => sb.Append(e.Data);
+                p.OutputDataReceived += (s, e) => { sb.Append(e.Data); if (--expectedLines == 0) are.Set(); };
```

Comment:
Do we guarantee that there will be a one-to-one correspondence between lines and OutputDataReceived invocations?